packages/valory/skills/task_execution/handlers.py

```diff
@@ -109,6 +110,15 @@ def handle(self, message: Message) -> None:
         dialogue = self.context.ipfs_dialogues.update(ipfs_msg)
         nonce = dialogue.dialogue_label.dialogue_reference[0]
         callback = self.params.req_to_callback.pop(nonce)
+        deadline = self.params.req_to_deadline.pop(nonce)
+        current_time = time.time()
+        if current_time > deadline:
+            self.context.logger.warning(
+                f" IPFS Message expired for {nonce}. Current time: {current_time}. Deadline: {deadline}"
+            )
+            self.params.in_flight_req = False
+            return
```

Comment:
the issue is that when the connection fails to send a response, which is why we are adding this logic in the first place,`handle` doesnt get called, it only gets called when the connection responds. 

I think this logic needs to be moved in the behaviour. `act()` gets called regardless, meaning these checks needs to be updated: https://github.com/valory-xyz/mech/blob/main/packages/valory/skills/task_execution/behaviours.py#L286.
In case this happens, I think we should just try to execute the task from scratch, in the same way we do when the task times out during execution.